plantcv/plantcv/visualize/time_lapse_video.py

```diff
@@ -0,0 +1,128 @@
+    :return: size_frame: tuple
+    """
+
+    ## Get the list of image files in the given directory and sort them alphabetically by their names
+    temp_list = []
+    for f in os.listdir(img_directory):
+        type_f = mimetypes.guess_type(f)[0]
+        if type_f:
+            if type_f.startswith('image'):
+                temp_list.append(f)
+    if len(temp_list) <= 0:
+        fatal_error("There is no file in the provided folder that is an image, please check the provided directory!")
+    else:
+#         temp_list.sort()
+        # get an "extension free" list of images
+        temp_list_no_ext = [os.path.splitext(f)[0] if os.path.splitext(f)[1] else f for f in temp_list]
+
+        ## if the list of images is provided, stick to it, but automatically handle the extension issue
+        if list_img is not None:
+            #  make the list of images extension free
+            list_img_no_ext = [os.path.splitext(f)[0] if os.path.splitext(f)[1] else f for f in list_img]
+            list_img_ = []
+            # check if the images in the given list exist in the directory by comparing both lists
+            for (f,f_) in zip(list_img_no_ext,list_img):
+                try:
+                    temp_list_no_ext.index(f)
+                    list_img_.append(f_)
+                except ValueError:
+                    continue
+            list_img = list_img_
+            if len(list_img) == 0:
+                fatal_error("The provided list of files is not contained in the given directory")
+            elif len(list_img) < len(list_img_no_ext):
+                warnings.warn("Warning: Some files in the provided list not found, the video will be created based on available files in the provided directory!")
+
+        # if the list of images is not provided, check if suffix information is available
+        elif suffix_img is not None:
+            list_img = [f for f in temp_list if f.endswith(suffix_img)]
+            if len(list_img) == 0:
+                fatal_error("There is no file that matches the provided suffix in the provided directory, please check again!")
+
+        # neither the list of image nor the suffix is provided
+        else:
+            list_img = temp_list
+        if auto_sort:
+            list_img.sort()
+
+        imgs   = []
+        list_r = []
+        list_c = []
+        for file in list_img:
+            img = cv2.imread(os.path.join(img_directory, file))
+            list_r.append(img.shape[0])
+            list_c.append(img.shape[1])
+            imgs.append(img)
+        max_c, max_r = np.max(list_c), np.max(list_r)
+
+        # If the frame size is not provided, use the largest size of the images as the frame size
+        size_frame = size_frame or (max_c, max_r)
+
+        if not (len(np.unique(list_r)) == 1 and len(np.unique(list_c)) == 1):
+            warnings.warn(f"The sizes of images are not the same, an image resizing (cropping or zero-padding) will be done "
```

Comment:
Is warnings.warn working as expected? Noah mentioned it only prints the warning in developing mode or something like that